pkg/apis/core/validation/validation.go

```diff
@@ -3873,9 +3873,35 @@ func ValidateContainerUpdates(newContainers, oldContainers []core.Container, fld
 			allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("image"), ctr.Image, "must not have leading or trailing whitespace"))
 		}
 	}
+
+	// validate updated container probe
+	for i, ctr := range newContainers {
```

Comment:
if probe.TerminationGracePeriodSeconds is still alpha, can we add the validation to require non-negative values before promotion to beta, rather than making the update validation more complicated/expensive?

Reply:
Sure. Update and I sent a new patch to follow this #103245